### ISSUE TITLE
Bool option is not parsed properly

### ISSUE THREAD

**Issue by vookimedlo** [MatthiWare/CommandLineParser.Core#35]
There is an issue with parsing the bool option without an explicit value.

```
    internal abstract class CommandOptions
    {
        [Name("v", "verb"), Description("Print usage"), DefaultValue(false)]
        public bool VerbA { get; set; }
    }

prog -v command
prog command -v
```

CommandOptions.VerrbA should be set to true (due to the -v), but is set to false. 

Bool resolver is correct and have the null value in a list of the expected 'true' values, but the code in upper layers is wrong.

```
namespace MatthiWare.CommandLine.Core.Parsing.Resolvers
{
    internal class BoolResolver : ArgumentResolver<bool>
    {
        private static readonly string[] recognisedFalseArgs = new[] { "off", "0", "false", "no" };
        private static readonly string[] recognisedTrueArgs = new[] { "on", "1", "true", "yes", string.Empty, null };
```

The main problem is in CommandLineCommand`TOption`TCommandOption.cs and CommandLineParser`TOption.cs.

```
                else if (!model.HasValue && option.HasDefault)
                {
                    option.UseDefault();

                    continue;
                }
```

and should be modified to something like this

```
                else if (!found && !model.HasValue && option.HasDefault)
                {
                    option.UseDefault();

                    continue;
                }
```

But that means, other resolvers shall be updated to cope with such situation and provide default value in such case.

**Comment by Matthiee** [comment]
Seems to be related to #30 can you see if that resolves it?

**Comment by vookimedlo** [comment]
No. The issue still exists on the most recent code from master.

**Comment by Matthiee** [comment]
You are correct bool options are behaving wrong when value `null` is provided and `DefaultValue(false)` is set. 

Your suggestion seems like a good fix but the other resolvers need to handle the situation accordingly. 

Do you want to take this one?

**Comment by vookimedlo** [comment]
Sorry, I don't have time for it now.

**Comment by Matthiee** [comment]
Thanks for letting me know, I'll take this one.

**Comment by vookimedlo** [comment]
I can confirm that it works in my case.

**Comment by Matthiee** [comment]
This ticket has been released in 0.2.2. 
https://www.nuget.org/packages/MatthiWare.CommandLineParser/0.2.2
